More selection rules

shaycrk opened the pull request:
Quick PR to add a few more selection rules:
* Fixes the variance-based rules that we didn't merge in with the initial set (`lowest_metric_variance` and `highest_avg_var_penalized`)
* Adds a rule to pick at random for a baseline (`random_model_group`)
* Adds a rule to pick based on the metric average with weights for recency (`highest_avg_recency_weight`)
* Renames `higest_metric_value` to `highest_current_value` for clarity

shaycrk committed:
random and min variance seln rules

thcrock committed:
Metric Directionality

- Change 'below_best' to 'dist_from_best_case' and 'raw_value' to 'dist_from_abs_worst' and make both dependent on the directionality of the metric according to catwalk, and test both types of metrics [Resolves #7]
- Add unit tests to selection rules [Resolves #5]

shaycrk committed:
var penalized and time weighted seln rules

shaycrk committed:
rename highest_metric_value to highest_current_value

shaycrk committed:
debugging

thcrock commented:
I am going to try and get test coverage in for these. Selection rules as we define them here are pretty easy to test, given the lack of external dependencies.  You'll see in the other PR I started this. These more complicated rules do require a bit more test data than the first round to capture their behavior, of course, but all the more reason we should do it.

shaycrk commented:
Yeah, now that I've had a chance to look at your PR, it definitely feels like we should try to bring that in first and then update this to both deal with directionality and include tests for the new selection rules before merging it.

thcrock committed:
First changes from review

thcrock committed:
Fix thresholding

thcrock committed:
Test cases for selection rules

thcrock committed:
Fix selection rules

thcrock committed:
Fix remaining var names/comments

thcrock committed:
Travis use postgres 9.5

shaycrk committed:
merging in metric_directionality

shaycrk committed:
update new rules for metric directionality

shaycrk committed:
add tests for new selection roles

shaycrk commented:
@thcrock -- merged in your `metric_directionality` branch here, updated the new rules to use directionality, and added tests for the new rules. However, I didn't add a test for the rule that returns a random model group since I didn't see a simple way to do so, but feel free to add one there, too, if you have an idea.

thcrock committed:
Merge branch 'master' into more_seln_rules

thcrock committed:
Fix merge problems
